src/Microsoft.CodeQuality.Analyzers/CSharp/QualityGuidelines/CSharpAvoidDuplicateElementInitialization.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpAvoidDuplicateElementInitialization : AvoidDuplicateElementInitialization
```

Comment:
Is there anything still causing the analyzer to be language/syntax specific? Ideally, you should be able to write this without Syntax APIs and move it to the core language agnostic layers.
```
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;
```

Reply:
I originally thought I had to break through to the syntax level to get the constant values, since `IArgumentOperation.ConstantValue` was never set, but after looking at it again, I see that `IArgumentOperation.Value.ConstantValue` holds what I need. I also found a way to avoid breaking through to the `AssignmentExpressionSyntax`. So yes, it looks like I could make this language agnostic. I'll send out another update soon.